src/state/slices/portfolioSlice/utils.ts

```diff
@@ -298,3 +298,33 @@ export const accountToPortfolio: AccountToPortfolio = args => {
 
   return portfolio
 }
+
+export const makeSortedAccountBalances = (totalAccountBalances: {
+  [k: AccountSpecifier]: string
+}) =>
+  Object.entries(totalAccountBalances)
+    .sort(([_, accountBalanceA], [__, accountBalanceB]) =>
+      bnOrZero(accountBalanceA).gte(bnOrZero(accountBalanceB)) ? -1 : 1
+    )
+    .map(([accountId, _]) => accountId)
+
+export const makeBalancesByChainBucketsFlattened = (
+  accountBalances: string[],
+  assets: { [k: CAIP19]: Asset }
+) => {
+  const balancesByChainBuckets = accountBalances.reduce(
+    (acc: Record<ChainTypes, CAIP10[]>, accountId) => {
+      const assetId = accountIdToFeeAssetId(accountId)
+      const asset = assets[assetId]
+
+      if (!acc[asset.chain]) {
+        acc[asset.chain] = []
+      }
+
+      acc[asset.chain] = [...acc[asset.chain], accountId]
+      return acc
+    },
+    {} as Record<ChainTypes, CAIP10[]>
+  )
```

Comment:
```suggestion
) => {

  const initial: Record<ChainTypes, CAIP10[]>
  const balancesByChainBuckets = accountBalances.reduce<Record<ChainTypes, CAIP10[]>>(
    (acc: Record<ChainTypes, CAIP10[]>, accountId) => {
      const assetId = accountIdToFeeAssetId(accountId)
      const asset = assets[assetId]
      acc[asset.chain] = [...(acc[asset.chain] ?? []), accountId]
      return acc
    },
    initial
  )
```

Reply:
a little bit of type safety and stylistic suggestions, but not required

Reply:
a little bit of type safety and stylistic suggestions, but not required

Reply:
Committed. Just went from typing of `initial` to a cast since the compiler isn't very happy with us passing an empty initial object as a `Record<ChainTypes, CAIP10[]>`